server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/AnalyticsServiceElasticsearch.java

```diff
@@ -740,28 +771,26 @@ private void buildCommunicationSummaryStatistics(Map<String, CommunicationSummar
 
             if (missingOp.getDocCount() > 0) {
                 Stats elapsed = missingOp.getAggregations().get("elapsed");
-                String id = urisBucket.getKey();
 
-                CommunicationSummaryStatistics css = stats.get(id);
+                CommunicationSummaryStatistics css = stats.get(uri);
                 if (css == null) {
                     css = new CommunicationSummaryStatistics();
-                    css.setId(id);
-                    css.setUri(id);
-
-                    String serviceName = serviceName(missingOp.getAggregations()
-                            .<Nested>get("nestedProperties").getAggregations()
-                            .<Filter>get("propertiesServiceFilter")
-                            .getAggregations().get("serviceTerm"));
-                    if (serviceName != null) {
-                        css.setServiceName(serviceName);
-                    }
-
-                    stats.put(id, css);
+                    css.setId(uri);
+                    css.setUri(uri);
+                    stats.put(uri, css);
```

Comment:
Do I understand it correctly that sometimes you put `id` as a key and sometimes `uri`?  (`id = EndpointUtil.encodeEndpoint(uri,
                          operationBucket.getKey());`)